morerss/zhihu_stream.py

```diff
@@ -100,27 +100,32 @@ class ZhihuAPI:
       'url': urljoin('https://www.zhihu.com/', url),
     }
 
-  async def topic_intro(self, name):
+  async def topic_info(self, id):
     """
     Zhihu topic information
-    :param name: Zhihu topic slug
-    :return:
+    :param id (str): Zhihu topic slug
+    :return (dict): dict containing the topic's name, description (omitted) and URL
     """
-    intro_url = 'https://www.zhihu.com/api/v4/topics/%s/intro' % name
-    res = await fetch_zhihu(
-      intro_url, headers={'User-Agent': self.user_agent})
-    if not res.body:
-      # e.g. https://www.zhihu.com/bei-feng-san-dai
+
+    url = urljoin('https://www.zhihu.com/topic/', id)
+
+    resp = await fetch_zhihu(
+      url, headers={'User-Agent': self.user_agent})
+    if not resp.body:
       raise web.HTTPError(404)
-    intro = json.loads(res.body.decode('utf-8'))
-    # topic name
-    name = intro['content']['meta']['name']
-    # topic description
-    description = intro['modules'][0]['target']['value']
-    url = urljoin('https://www.zhihu.com/topic', name)
+    doc = fromstring(resp.body.decode('utf-8'))
+
+    if doc.xpath('//div[@class="TopicMetaCard-title"]'):
+      name = doc.xpath('//div[@class="TopicMetaCard-title"]')[0].text_content()
+    elif doc.xpath('//h1[@class="TopicCard-titleText"]'):
+      name = doc.xpath('//h1[@class="TopicCard-titleText"]')[0].text_content()
+    else:
+      name = id
+
     return {
       'name': name,
-      'description': description,
+      # Omit the description
+      'description': '',
```

Comment:
Why is it always omitted? If it's not always available, do not use it.

Reply:
637bfda adds `description` field.